From 4605fa4ea42cedb223730a287ee5c532eac681ea Mon Sep 17 00:00:00 2001
From: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Date: Tue, 4 Feb 2025 15:18:08 +0100
Subject: [PATCH 1513/1695] dts: stm32: update dma range for the mlhab node for
 stm32mp2x

Declare alias regions 0x000000000 and 0x100000000 used for the
support of the Cortex-M33 Icache.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git a9c7e0c6d97e01351ad19cdcdc7669e7b7ed30b2 dts: stm32: update dma range for the mlhab node for stm32mp2x]
Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Change-Id: I683a14339d4223cb4de00847d6cf24ec7755917f
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/434526
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Tested-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
---
 core/arch/arm/dts/stm32mp211.dtsi | 6 ++++--
 core/arch/arm/dts/stm32mp231.dtsi | 5 ++++-
 core/arch/arm/dts/stm32mp251.dtsi | 5 ++++-
 3 files changed, 12 insertions(+), 4 deletions(-)

diff --git a/core/arch/arm/dts/stm32mp211.dtsi b/core/arch/arm/dts/stm32mp211.dtsi
index dccd64c59..4e04eb96d 100644
--- a/core/arch/arm/dts/stm32mp211.dtsi
+++ b/core/arch/arm/dts/stm32mp211.dtsi
@@ -121,8 +121,10 @@
 		#address-cells = <1>;
 		#size-cells = <1>;
 		ranges = <0x0 0x0 0x0 0xFFFFFFFC>;
-		dma-ranges = <0x0 0x0 0x0 0xFFFFFFFC>;
-
+		dma-ranges = <0x00000000 0x0 0x80000000 0x08000000>,
+			     <0x08000000 0x0 0x08000000 0x08000000>,
+			     <0x10000000 0x0 0x80000000 0x10000000>,
+			     <0x20000000 0x0 0x20000000 0xD0000000>;
 		m33_rproc: m33@0 {
 			compatible = "st,stm32mp2-m33";
 			#reset-cells = <1>;
diff --git a/core/arch/arm/dts/stm32mp231.dtsi b/core/arch/arm/dts/stm32mp231.dtsi
index f9eb73ee9..1fc045f30 100644
--- a/core/arch/arm/dts/stm32mp231.dtsi
+++ b/core/arch/arm/dts/stm32mp231.dtsi
@@ -123,7 +123,10 @@
 		#address-cells = <1>;
 		#size-cells = <1>;
 		ranges = <0x0 0x0 0x0 0xFFFFFFFC>;
-		dma-ranges = <0x0 0x0 0x0 0xFFFFFFFC>;
+		dma-ranges = <0x00000000 0x0 0x80000000 0x08000000>,
+			     <0x08000000 0x0 0x08000000 0x08000000>,
+			     <0x10000000 0x0 0x80000000 0x10000000>,
+			     <0x20000000 0x0 0x20000000 0xD0000000>;
 
 		m33_rproc: m33@0 {
 			compatible = "st,stm32mp2-m33";
diff --git a/core/arch/arm/dts/stm32mp251.dtsi b/core/arch/arm/dts/stm32mp251.dtsi
index 318bb96bb..c12077835 100644
--- a/core/arch/arm/dts/stm32mp251.dtsi
+++ b/core/arch/arm/dts/stm32mp251.dtsi
@@ -123,7 +123,10 @@
 		#address-cells = <1>;
 		#size-cells = <1>;
 		ranges = <0x0 0x0 0x0 0xFFFFFFFC>;
-		dma-ranges = <0x0 0x0 0x0 0xFFFFFFFC>;
+		dma-ranges = <0x00000000 0x0 0x80000000 0x08000000>,
+			     <0x08000000 0x0 0x08000000 0x08000000>,
+			     <0x10000000 0x0 0x80000000 0x10000000>,
+			     <0x20000000 0x0 0x20000000 0xD0000000>;
 
 		m33_rproc: m33@0 {
 			compatible = "st,stm32mp2-m33";
-- 
2.47.2

